chall05/vkuikka.rb

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/ruby
+#ruby 2.5.5p157
+
+require 'open-uri'
+
+if (!article = ARGV[0])
+	return
+end
+url = "https://en.wikipedia.org/wiki/"
+begin
+file = open(url + article)
+rescue
+	puts "No wikipedia page for this word"
+	return
+end
+visited = []
+while (article != "Philosophy")
+	visited << article
+	file = open(url + article)
+	content = file.read
+	matches = content.scan(/(?<=(<p>)|(<h[0-9])).+?<a href=\"\/wiki\/([^\":]*)"/mu).drop(1)
+	for items in matches
+		if !visited.include? items[2]
```

Comment:
even if Ruby allows it, I would recommend keeping writing () to pass parameters.